cmake/CompileBoost.cmake

```diff
@@ -145,7 +145,7 @@ if(WIN32)
   # properly for config mode. So we use the old way on Windows
   #  find_package(Boost 1.72.0 EXACT QUIET REQUIRED CONFIG PATHS ${BOOST_HINT_PATHS})
   # I think depending on the cmake version this will cause weird warnings
-  find_package(Boost 1.72 COMPONENTS filesystem iostreams)
+  find_package(Boost 1.78 COMPONENTS filesystem iostreams serialization system)
```

Comment:
Not sure why 1.78 is needed for windows build? Upgrading boost may require build changes, so I asked @ammolitor to take a look as well.

Reply:
Mainly for consistency with Linux builds which use 1.78 as well. The 1.72 version is just as minimum; the CI always used 1.78 for Windows builds